src/engine/x86-64/X86_64Interpreter.v3

```diff
@@ -2220,6 +2220,38 @@ class X86_64InterpreterGen(ic: X86_64InterpreterCode, w: DataWriter) {
 		asm_mov_m_r(mem_addr, val);
 		endHandler();
 	}
+	def gen_i8X16_splat() {
+		masm.emit_i8x16_splat(r_xmm0, r_tmp0, r_xmm1);
+	}
+	def gen_i16X8_splat() {
+		masm.emit_i16x8_splat(r_xmm0, r_tmp0);
+	}
+	def gen_i32X4_splat() {
+		masm.emit_i32x4_splat(r_xmm0, r_tmp0);
+	}
+	def gen_i64X2_splat() {
+		masm.emit_i64x2_splat(r_xmm0, r_tmp0);
+	}
+	def gen_f32X4_splat() {
+		asm.pinsrd_s_r_i(r_xmm1, r_tmp0, 0);
+		masm.emit_f32x4_splat(r_xmm0, r_xmm1);
+	}
+	def gen_f64X2_splat() {
+		asm.pinsrq_s_r_i(r_xmm1, r_tmp0, 0);
+		masm.emit_f64x2_splat(r_xmm0, r_xmm1);
+	}
+	def genSplat(opcode: Opcode, from_memarg: bool,
```

Comment:
If you further split this method into two versions (`from_memarg=true` and `from_memarg=false`), then you can just pass methods from the `MacroAssembler` to it.

Reply:
You can just bind a temporary register for the one that takes a scratch register, e.g. `masm.emit_i8x16_splat(_, _, r_tmp2)`.

Reply:
Yeah I just forgot to use the partial application feature of Virgil. It's so convenient. Pushed a new change.